backend/curation/api/v1/curation/collections/collection_id/datasets/dataset_id/manifest/actions.py

```diff
@@ -12,16 +12,50 @@
     _get_collection_and_dataset,
 )
 from backend.layers.auth.user_info import UserInfo
+from backend.layers.business.business import BusinessLogic
 from backend.layers.business.exceptions import (
     CollectionIsPublishedException,
     CollectionNotFoundException,
     DatasetInWrongStatusException,
     DatasetNotFoundException,
     InvalidURIException,
 )
+from backend.layers.common.entities import DatasetArtifactType
 from backend.portal.api.providers import get_business_logic
 
 
+def get_single_artifact_permanent_url(dataset_version, artifact_type, required=False) -> str:
+    """Find exactly one artifact of the given type, then return it's canonical URI.
+
+    If `required` is True and no artifact is found, raises ValueError.
+    If more than one is found, always raises ValueError.
+    """
+    artifacts = dataset_version.artifacts
+    matches = [a for a in artifacts if a.type == artifact_type]
+
+    if len(matches) > 1:
+        raise ValueError(f"Multiple '{artifact_type}' artifacts found.")
+
+    if not matches and required:
+        raise ValueError(f"No '{artifact_type}' artifact found.")
+
+    return BusinessLogic.generate_permanent_url(dataset_version.version_id, artifact_type) if matches else None
```

Comment:
The atac fragment will not be stored on the `dataset_version.version_id` but instead under the artifact_id. We will need to handle that case when returning The permanent url.